sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -64,6 +65,22 @@ class SparkSession private(
    |  Session-related state  |
    * ----------------------- */
 
+  {
+    val defaultWarehousePath =
+      SQLConf.WAREHOUSE_PATH
+        .defaultValueString
+        .replace("${system:user.dir}", System.getProperty("user.dir"))
+    val warehousePath = sparkContext.conf.get(
+      SQLConf.WAREHOUSE_PATH.key,
+      defaultWarehousePath)
+    sparkContext.conf.set(SQLConf.WAREHOUSE_PATH.key, warehousePath)
+    sparkContext.conf.set("hive.metastore.warehouse.dir", warehousePath)
+    // scalastyle:off println
+    println(s"Setting warehouse location to $warehousePath")
```

Comment:
Sure. Will it be useful to let users know the warehouse location when they start a spark shell?